Improve debug_ipint.py

ast-hugger opened the pull request:
#### db255dd395acdf767640ec368c7a6affb9dabc89
<pre>
Improve debug_ipint.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=293443">https://bugs.webkit.org/show_bug.cgi?id=293443</a>
<a href="https://rdar.apple.com/151868845">rdar://151868845</a>

Reviewed by Daniel Liu.

This patch fixes some outstanding issues in debug_ipint.py, some introduced by an earlier set of
changes, some original:

    - ipint_break_at command
    - ipint_stack command

It also somewhat organizes the instruction symbol list and provides a tentative future change
(commented out) for the renaming of `fb_block` and 3 other instructions that was attempted
by <a href="https://bugs.webkit.org/show_bug.cgi?id=292725">https://bugs.webkit.org/show_bug.cgi?id=292725</a> but has been reverted for now.

* Source/JavaScriptCore/llint/InPlaceInterpreter64.asm:
* Tools/lldb/debug_ipint.py:
(extract_gprs):
(print_value):
(print_stack):
(find_instruction_addresses):
(find_instruction):
(ipint_state):
(ipint_stack):
(ipint_local):
(ipint_continue_until):
(set_breakpoints_internal):
(ipint_break_at):
(ipint_disable_all_breakpoints):
(ipint_reenable_all_breakpoints):
(ipint_continue_on_all_breakpoints):
(set_breakpoints):
(get_load_address): Deleted.
(get_instruction_addresses): Deleted.

Canonical link: <a href="https://commits.webkit.org/295340@main">https://commits.webkit.org/295340@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4bde8308f1262cd5bc06d563d0f55ef25ac2850e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104703 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24415 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14837 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109917 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55377 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106743 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24806 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32961 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79511 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107709 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19300 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94503 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59818 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19067 "layout-tests (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12578 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54752 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/97384 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88764 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12628 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112312 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/103321 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31868 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23445 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88590 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32232 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90732 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88208 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22498 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33102 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10881 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/27181 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31793 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37146 "Built successfully") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/127600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31585 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/127600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34926 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33144 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dhecht commented on Tools/lldb/debug_ipint.py:
```diff
@@ -27,45 +27,55 @@
+    'ref_null_t', 'ref_is_null', 'ref_func', 'ref_eq', 'ref_as_non_null', 'br_on_null', 'br_on_non_null',
+    # 0xFB
+    'fb_block', 'fc_block', 'simd', 'atomic',
+    # The above line after the "Harden IPInt dispatch" PR, reverted now: https://bugs.webkit.org/show_bug.cgi?id=292725
```
I'm not sure what this means. Do you mean the line below? And is it saying something should happen in the future (FIXME)?

danlliu replied:
+1, I believe it's referring to the line below and to add these back when that PR is re-landed.

ast-hugger replied:
Oops, yes, I did mean the line below. There was a PR tracked by that bugzilla which renamed 'fb_block' and other instructions on this line to what's commented out below. That changed symbol names and so instruction lookup was failing in my branch that had the PR in. Then I checked with  the current tip of main and found the PR was reverted, but I'm assuming it's coming back. I'll fix the comment.

dhecht reviewed (commented):
LGTM (non-binding), just not sure about that comment.

danlliu reviewed (approved):
r=me with added FIXME as mentioned by Dan

ast-hugger committed:
Improve debug_ipint.py
https://bugs.webkit.org/show_bug.cgi?id=293443
rdar://151868845

Reviewed by Daniel Liu.

This patch fixes some outstanding issues in debug_ipint.py, some introduced by an earlier set of
changes, some original:

    - ipint_break_at command
    - ipint_stack command

It also somewhat organizes the instruction symbol list and provides a tentative future change
(commented out) for the renaming of `fb_block` and 3 other instructions that was attempted
by https://bugs.webkit.org/show_bug.cgi?id=292725 but has been reverted for now.

* Source/JavaScriptCore/llint/InPlaceInterpreter64.asm:
* Tools/lldb/debug_ipint.py:
(extract_gprs):
(print_value):
(print_stack):
(find_instruction_addresses):
(find_instruction):
(ipint_state):
(ipint_stack):
(ipint_local):
(ipint_continue_until):
(set_breakpoints_internal):
(ipint_break_at):
(ipint_disable_all_breakpoints):
(ipint_reenable_all_breakpoints):
(ipint_continue_on_all_breakpoints):
(set_breakpoints):
(get_load_address): Deleted.
(get_instruction_addresses): Deleted.

Canonical link: https://commits.webkit.org/295340@main
